concordia/views.py

```diff
@@ -221,26 +221,23 @@ def post(self, *args, **kwargs):
             cmd = 'cp -r {0}/* {1}'.format('/concordia_images', collection_path)
             if os.WEXITSTATUS(os.system(cmd)) == 0:
                 os.system('rm -rf {0}'.format('/concordia_images/*'))
-            c = Collection.objects.create(title=name, slug=name.replace(" ", "-"), description=name)
-            count = 0
-            for root, dirs, files in os.walk(collection_path):
-                for filename in files:
-                    filename = os.path.join(root, filename)
-                    if True:
-                        count += 1
-                        title = '{0} asset {1}'.format(name, count)
-                        media_url = os.path.join(root, filename).replace(settings.MEDIA_ROOT, '')
+                c = Collection.objects.create(title=name, slug=name.replace(" ", "-"), description=name)
+                for root, dirs, files in os.walk(collection_path):
```

Comment:
This is enough logic that it seems like we should have it broken into separate functions / methods so the view would clearly communicate the high-level process:

```python

c = Collection.objects.create(…)
c.copy_images_to_collection(…)
c.create_assets_from_filesystem(…)
```

Reply:
One advantage to that approach would be highlighting bugs such as the race condition where two simultaneous requests would intermix their files in `/concordia_images` rather than correctly separating them.